packages/ingest-mongodb-public/src/sources/WebDataSource.ts

```diff
@@ -0,0 +1,352 @@
+import { Page } from "mongodb-rag-core";
+import { DataSource } from "mongodb-rag-core/dataSources";
+import * as cheerio from "cheerio";
+import puppeteer, { Page as PuppeteerPage } from "puppeteer";
+import TurndownService from "turndown";
+import * as turndownPluginGfm from "turndown-plugin-gfm";
+import xml2js from "xml2js";
+
+// TODO: move to constants file
+const sitemapURL = "https://www.mongodb.com/sitemap-pages.xml";
+// urls of the directories where we want every single page
+const directoryUrls = [
+  "https://www.mongodb.com/solutions/customer-case-studies",
+  "https://www.mongodb.com/solutions/solutions-library/",
+];
+const individualUrls = [
```

Comment:
all of this implementation specific detail should be passed as config to `makeWebDataSource()`